### ISSUE TITLE
Width computation

### ISSUE THREAD

**Issue by manu-st** [bpampuch/pdfmake#1536]
Hi

I have wide tables and they are truncate when they are generated.I saw that when I call `createPdfKitDocument` it updates my document definition with `_minWidth`, `_maxWidth` and `_calcWidth`. It seems that based on that info I could easily figure out at which column I need to truncate my table so it is never truncated.

Is there a way to compute them before calling `createPdfKitDocument`?

Thanks,
Manu

**Comment by manu-st** [comment]
Also if you could clarify the meaning of `_minWidth`, `_maxWdith` and `_calcWidth` as they seem to always end up having the same value in my tests.

**Comment by liborm85** [comment]
`_minWidth`, `_maxWidth`, `_calcWidth` and etc. are internal variables for pdf document rendering. If they can help you, you can use them.
It is not possible calculate before calling pdfmake.

Wide tables are problematic and feature request is available here: https://github.com/bpampuch/pdfmake/issues/441

**Comment by manu-st** [comment]
I managed to generate wide tables by calling createPdfKitDocument with my table definition. Then based on the computed widths for the table and the offsets, I figured out where I should split my tables. Once I had the split positions, I simply regenerate as many table as I needed so that it fits the width of my page.

It is a two step process but seems to work quite nicely.

If this can be done by pdfMake internally, that would be great.

Closing since I have a solution that works.
